### ISSUE TITLE
`per-file-ignores` with ALL creates a false positive

### ISSUE THREAD

**Issue by tooruu** [astral-sh/ruff#9297]
Given the following Python file
```py
# T201 disallows `print`
print("foo")  # noqa: T201
```
and the following Ruff configuration
```toml
select = ["ALL"]

[per-file-ignores]
"file.py" = ["ALL"]
```
`ruff check` will still raise error ``RUF100 [*] Unused `noqa` directive (unused: `T201`)``.

**Comment by tooruu** [comment]
I'm trying to exclude files from linting but not formatting, without editing those files.

**Comment by charliermarsh** [comment]
Seems like a bug as-described, will take a look.

**Comment by charliermarsh** [comment]
We also support a dedicated exclude keyword under the lint namespace which will only exclude files from linting and is the preferred solution for what you’re describing, eg:

[lint]
exclude = [“file.py”]

**Comment by tooruu** [comment]
> We also support a dedicated exclude keyword under the lint namespace which will only exclude files from linting and is the preferred solution for what you’re describing, e.g.:
> 
> ```toml
> [lint]
> exclude = ["file.py"]
> ```

Didn't realize it was under lint namespace! Thank you.

**Comment by charliermarsh** [comment]
Regardless, it _is_ a bug so fixed in the next release :) Thanks!
